src/panels/ShareModelPanel/ShareModel.tsx

```diff
@@ -39,28 +124,232 @@ export default function ShareModel() {
                 <div className="share-model__card" key={userObj.user}>
                   <div className="share-model__card-title">
-                    <strong>{userObj["user"]}</strong>
+                    <strong>{userName}</strong>
                     <span className="secondary">
-                      {isOwner(userObj["user"]) && "Owner"}
+                      {isOwner(userName) ? (
+                        "Owner"
+                      ) : (
+                        <i
+                          className="p-icon--delete"
+                          onClick={() => handleRemoveUser(userName)}
+                          onKeyPress={() => handleRemoveUser(userName)}
+                          role="button"
+                          tabIndex={0}
+                        >
+                          Remove user
+                        </i>
+                      )}
                     </span>
                   </div>
-                  <p className="supplementary">
+                  <div className="supplementary">
                     Last connected:{" "}
-                    {userObj["last-connection"]
-                      ? formatFriendlyDateToNow(userObj["last-connection"])
+                    {lastConnected
+                      ? formatFriendlyDateToNow(lastConnected)
                       : `Never connected`}
-                  </p>
+                    <Formik
+                      initialValues={{
+                        selectAll: false,
+                        selectedUnits: [],
+                      }}
+                      onSubmit={() => {}}
+                    >
+                      <FormikFormData>
+                        <Field
+                          as="select"
+                          name="accessLevel"
+                          onChange={async (
+                            e: React.ChangeEvent<HTMLInputElement>
+                          ) => {
+                            const cloneUserAccess = cloneDeep(usersAccess);
+                            if (cloneUserAccess) {
+                              cloneUserAccess[userName] = e.target.value;
+                            }
+                            setUsersAccess(cloneUserAccess);
+                            const updatedUserAccess: userAccess = {
+                              name: userName,
+                              access: e.target.value,
+                            };
+                            const response = await setModelSharingPermissions(
+                              modelControllerURL,
+                              modelUUID,
+                              store.getState,
+                              updatedUserAccess,
+                              usersAccess?.[userName],
+                              "grant",
+                              dispatch
+                            );
+                            const error = response?.results[0]?.error?.message;
+                            if (error) {
```

Comment:
Is it possible to error now that you have the UI setup correctly?

Reply:
Yes, the API returns any error from the grant/revoke calls here.